src/include/misc.h

```diff
@@ -274,10 +274,24 @@
  */
 #define WT_STREQ(s, cs) (sizeof(cs) == 2 ? (s)[0] == (cs)[0] && (s)[1] == '\0' : strcmp(s, cs) == 0)
 
-/* Check if a string matches a byte string of len bytes. */
-#define WT_STRING_MATCH(str, bytes, len)                                                        \
-    (((const char *)(str))[0] == ((const char *)(bytes))[0] && strncmp(str, bytes, len) == 0 && \
-      (str)[len] == '\0')
+/*
+ * Check if a string matches a byte string of len bytes. This macro expects both strings to be of
+ * non-zero length in order to be a match - this means that strings cannot be a match if len
+ * == 0.
+ *
+ * Use an inline function as that tends to be safer and we can check for NULL values.
+ */
+#define WT_STRING_MATCH(str, bytes, len) __wt_string_match(str, bytes, len)
```

Comment:
As discussed would there be value in have a `WT_STRING_MATCH_CVAL(str, cval)` as a convenience?

Reply:
I would rather keep it as one macro as it seems cleaner that way.